nuxeo-coldstorage/src/main/java/org/nuxeo/coldstorage/events/CheckColdStorageContentRestoreListener.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * (C) Copyright 2021 Nuxeo (http://nuxeo.com/) and others.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * Contributors:
+ *     Abdoul BA<aba@nuxeo.com>
+ */
+
+package org.nuxeo.coldstorage.events;
+
+import static org.nuxeo.coldstorage.events.CheckUpdateMainContentInColdStorage.DISABLE_COLD_STORAGE_CHECK_UPDATE_MAIN_CONTENT_LISTENER;
+import static org.nuxeo.coldstorage.helpers.ColdStorageHelper.COLD_STORAGE_CONTENT_PROPERTY;
+import static org.nuxeo.coldstorage.helpers.ColdStorageHelper.COLD_STORAGE_CONTENT_RESTORED_EVENT_NAME;
+import static org.nuxeo.coldstorage.helpers.ColdStorageHelper.COLD_STORAGE_FACET_NAME;
+import static org.nuxeo.coldstorage.helpers.ColdStorageHelper.COLD_STORAGE_TO_BE_RESTORED_PROPERTY;
+import static org.nuxeo.coldstorage.helpers.ColdStorageHelper.FILE_CONTENT_PROPERTY;
+
+import java.io.Serializable;
+
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.event.EventBundle;
+import org.nuxeo.ecm.core.event.EventService;
+import org.nuxeo.ecm.core.event.PostCommitEventListener;
+import org.nuxeo.ecm.core.event.impl.DocumentEventContext;
+import org.nuxeo.runtime.api.Framework;
+
+/**
+ * An asynchronous listener that checks if the contents already retrieved from cold storage can be restored.
+ *
+ * @apiNote: This listener is designed to be called from a scheduler.
```

Comment:
This listener is not called from the scheduler itself but from the helper. And, can the event can be dispatched from anywhere.